lib/package_manager.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+function package_manager::determine_package_manager() {
+	local build_dir="${1}"
+	local package_managers_found=()
+
+	if [[ -f "${build_dir}/Pipfile.lock" ]]; then
+		package_managers_found+=(pipenv)
+		meta_set "pipenv_has_lockfile" "true"
+	elif [[ -f "${build_dir}/Pipfile" ]]; then
+		# TODO: Start requiring a Pipfile.lock and make this branch a "missing lockfile" error instead.
+		package_managers_found+=(pipenv)
+		meta_set "pipenv_has_lockfile" "false"
+	fi
+
+	if [[ -f "${build_dir}/requirements.txt" ]]; then
+		package_managers_found+=(pip)
+	fi
+
+	# TODO: Deprecate/sunset this fallback, since using setup.py declared dependencies is
+	# not a best practice, and we can only guess as to which package manager to use.
+	if ((${#package_managers_found[@]} == 0)) && [[ -f "${build_dir}/setup.py" ]]; then
+		package_managers_found+=(pip)
+		meta_set "setup_py_only" "true"
+	else
+		meta_set "setup_py_only" "false"
+	fi
+
+	case "${#package_managers_found[@]}" in
+		1)
+			echo "${package_managers_found[0]}"
+			return 0
+			;;
+		0)
+			display_error <<-EOF
+				Error: Couldn't find any supported Python package manager files.
+
+				A Python app on Heroku must have either a 'requirements.txt' or
+				'Pipfile' package manager file in the root directory of its
+				source code.
+
+				Currently the root directory of your app contains:
+
+				$(ls -1 --indicator-style=slash "${build_dir}" || true)
+
+				If your app already has a package manager file, check that it:
+
+				1. Is in the top level directory (not a subdirectory).
+				2. Has the correct spelling (the filenames are case-sensitive).
+				3. Isn't listed in '.gitignore' or '.slugignore'.
+
+				Otherwise, add a package manager file to your app. If your app has
+				no dependencies, then create an empty 'requirements.txt' file.
+
+				For help with using Python on Heroku, see:
+				https://devcenter.heroku.com/articles/getting-started-with-python
+				https://devcenter.heroku.com/articles/python-support
```

Comment:
issue: replace with Scalingo documentation page

Reply:
Fixed in https://github.com/Scalingo/python-buildpack/pull/115/commits/76f3121da76b31b4cbd9f45aa858b5c2f2311e10

Reply:
Thanks :)